ehcache-core/src/main/java/org/ehcache/core/EhcachePrefixLoggerFactory.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.core;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Factory to create instances of {@link PrefixLogger}.
+ *
+ * A thread can call {@link #withContext(String, String)} method to set the context key-value pair
+ *
+ * <pre>{@code
+ *     try(EhcachePrefixLoggerFactory.Context ignored = EhcachePrefixLoggerFactory.withContext("cache-name", "my-cache")){
```

Comment:
I would avoid using cache-name as the example here. The context could be anything... let use something more generic.

Reply:
changed to `withContext("cache-key1", "cache-value1")`